packages/core/src/lib/session.ts

```diff
@@ -41,6 +42,30 @@ export const assignInteractionResults = async (
   ctx.body = { redirectTo, ts };
 };
 
+export const checkSessionHealth = async (
+  ctx: Context,
+  provider: Provider,
+  tolerance = 10 * 60 // 10 mins
+) => {
+  const { result } = await provider.interactionDetails(ctx.req, ctx.res);
+
+  if (!result?.login?.accountId) {
+    throw new RequestError('auth.unauthorized');
+  }
+
+  if (!result.login.ts || result.login.ts < dayjs().unix() - tolerance) {
```

Comment:
maybe need to change the usage of `dayjs` to `date-fns` as Gao mentioned.

Reply:
done